src/sssom/io.py

```diff
@@ -147,7 +164,9 @@ def get_metadata_and_prefix_map(
         metadata = yaml.safe_load(file)
 
     metadata = dict(ChainMap(metadata, get_default_metadata()))
-    converter = Converter.from_prefix_map(metadata.pop(CURIE_MAP, {}))
+    prefix_map = metadata.pop(CURIE_MAP, {})
+    epm = convert_plain_prefix_map_to_extended(prefix_map)
+    converter = Converter.from_extended_prefix_map(epm)
```

Comment:
## Update: `get_metadata_and_prefix_map()`
<details><summary>Details</summary>
<p>

### Overview
Now instantiates a `Converter` from extended prefix map (EPM) instead of plain prefix map.

To see how this works, look at my comment for `convert_plain_prefix_map_to_extended()`.

It does this in a funky way which I think needs rethinking and refactoring. But that could take a fair amount of discussion. For now, since the Mondo pipeline is completely broken by this, I chose to make this quicker change.

## Other considerations & blockers


</p>
</details>

Reply:
> Can we rename this? Because that's what it actually does: return converter, metadata

I prefer to only rename things if they are either _private_ methods, or there is no other way to not create churn for downstream users.

Reply:
Added deprecation PR for this in https://github.com/mapping-commons/sssom-py/pull/486

Reply:
That's what I was gonna suggest. Rename it to `_get_converter_and_metadata()`, have the old function reroute to this one, and deprecate the old function.

Reply:
Charlie has addressed this.

Other functionality I added has now been removed. It is not necessary to solve any problems in https://github.com/monarch-initiative/mondo-ingest/pull/394. It can be re-handled in some variation in future PR for EPM support.``